--- /dev/null
+++ b/libs/utils/Makefile
@@ -0,0 +1,63 @@
+#!/bin/make
+
+CXX = g++
+
+SOURCES:= \
+	BasicHashtable.cpp \
+	BlobCache.cpp \
+	BufferedTextOutput.cpp \
+	Debug.cpp \
+	FileMap.cpp \
+	Flattenable.cpp \
+	LinearTransform.cpp \
+	Looper.cpp \
+	PropertyMap.cpp \
+	RefBase.cpp \
+	SharedBuffer.cpp \
+	Static.cpp \
+	StopWatch.cpp \
+	String8.cpp \
+	String16.cpp \
+	StringArray.cpp \
+	SystemClock.cpp \
+	TextOutput.cpp \
+	Threads.cpp \
+	Timers.cpp \
+	Tokenizer.cpp \
+	Unicode.cpp \
+	VectorImpl.cpp \
+	WorkQueue.cpp \
+	ZipFileCRO.cpp \
+	ZipFileRO.cpp \
+	ZipUtils.cpp \
+	misc.cpp
+
+OBJECTS = $(SOURCES:.cpp=.o)
+
+# Debian has gettid, but this build doesn't find it
+#         -DHAVE_GETTID \
+
+ALL_CXXFLAGS = -O2 -fPIC -I../../include -I/usr/include/android \
+	-include /usr/include/android/arch/linux-x86/AndroidConfig.h
+ALL_LDFLAGS = -fPIC -shared -rdynamic
+ALL_LIBS = -lrt -ldl -lz -lpthread -L/usr/lib/android -lcutils -llog
+
+CXXFLAGS := $(ALL_CXXFLAGS) $(CXXFLAGS)
+LDFLAGS := $(ALL_LDFLAGS) $(LDFLAGS)
+LIBS := $(ALL_LIBS) $(LIBS)
+
+LIBNAME = libutils
+
+all: $(OBJECTS) $(LIBNAME).pc
+	$(CXX) $(LDFLAGS) -Wl,-soname,$(LIBNAME).so.0 -o $(LIBNAME).so.0 $(OBJECTS) $(LIBS)
+	ln -sf $(LIBNAME).so.0 $(LIBNAME).so
+	ar rs $(LIBNAME).a $(OBJECTS)
+
+$(LIBNAME).pc: $(LIBNAME).pc.in
+	sed -e 's/@VERSION@/$(VERSION)/g' -e 's/@TRIPLET@/$(TRIPLET)/g' $(LIBNAME).pc.in > $(LIBNAME).pc
+
+clean:
+	rm -f $(OBJECTS)
+	rm -f $(LIBNAME).so $(LIBNAME).so.0 $(LIBNAME).a
+	rm -f $(LIBNAME).pc
+
--- /dev/null
+++ b/libs/utils/libutils.pc.in
@@ -0,0 +1,13 @@
+prefix=/usr
+exec_prefix=
+libdir=${prefix}/lib/@TRIPLET@
+includedir=${prefix}/include/android
+
+Name: libutils
+Description: Android native utils library
+Version: @VERSION@
+
+Requires: androidlog = @VERSION@, bsd
+Libs: -L${libdir} -lutils
+Libs.private: -lpthread
+Cflags: -I${includedir}
